### PR TITLE
[FIX] stock: create SM from SML

### PR DESCRIPTION
**Pull request by pgu-odoo**
In this commit, creating SM from SML if trasfer is immediate. Due
to which following issues are getting resolve -
For immediate trasfer -

1) Unable to cancel the picking in draft state and SML is created
by view directly (No SM).

2) Unable to see button Put in pack.

3) Picking remains in state draft instead of ready.

task-2275833

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/stock/models/stock_move_line.py [hunk 1]
```diff
@@ -173,21 +174,29 @@ def create(self, vals_list):
...
                         'additional': True,
                         'picking_id': picking.id,
+                        'restrict_partner_id': picking.owner_id.id,
+                        'company_id': picking.company_id.id,
                     })
+                    if picking.state != 'done':
+                        new_move._action_confirm()
                     vals['move_id'] = new_move.id
-        mls = super(StockMoveLine, self).create(vals_list)
+            mls |= super(StockMoveLine, self).create(vals)
```
Comment (amoyaux) [inline addons/stock/models/stock_move_line.py:199]: you could probably keep it in batch

File: addons/stock/models/stock_picking.py [hunk 2]
```diff
@@ -515,7 +515,7 @@ def _compute_show_check_availability(self):
     @api.depends('state', 'move_lines')
     def _compute_show_mark_as_todo(self):
         for picking in self:
-            if not picking.move_lines and not picking.package_level_ids:
+            if not picking.package_level_ids and picking.immediate_transfer:
```
Comment (amoyaux) [inline addons/stock/models/stock_picking.py:518]: Why do you need this change?

Comment (pgu-odoo) [reply addons/stock/models/stock_picking.py:518]: yes, its working without this change. Removing it.

File: addons/stock/models/stock_move_line.py (hunk 1 above)
Comment (pgu-odoo) [reply addons/stock/models/stock_move_line.py:199]: Can you please elaborate? I didn't get what you mean.

Comment (amoyaux) [reply addons/stock/models/stock_move_line.py:199]: you could keep it as before.

File: addons/stock/models/stock_move_line.py [hunk 3]
```diff
@@ -163,6 +163,7 @@ def _onchange_qty_done(self):
 
     @api.model_create_multi
     def create(self, vals_list):
+        mls = self.env['stock.move.line']
```
Comment (amoyaux) [inline addons/stock/models/stock_move_line.py:166]: not needed anymore

**Comment by sle-odoo** [comment]
@pgu-odoo branch is red

**Comment by pgu-odoo** [comment]
> @pgu-odoo branch is red
Its green now.

**Comment by sle-odoo** [comment]
@pgu-odoo rebase is failed

**Comment by sle-odoo** [comment]
robodoo r+ rebase-ff

**Comment by amoyaux** [comment]
@pgu-odoo It seems that it break enterprise tests. Could you checK?

**Comment by pgu-odoo** [comment]
> @pgu-odoo It seems that it break enterprise tests. Could you checK?
ok I am checking.

**Comment by d-fence** [comment]
robodoo retry

**Comment by d-fence** [comment]
robodoo r-

**Comment by Xavier-Do** [comment]
@robodoo r- this one is a legit failure I think, can you check?

**Comment by Whenrow** [comment]
see https://github.com/odoo/odoo/pull/57098 to retarget on saas-13.5

**Pull request closed without merging**